include/oneapi/dpl/pstl/tuple_impl.h

```diff
@@ -256,14 +298,49 @@ struct tuple<T1, T...>
 
     using tuple_type = ::std::tuple<T1, T...>;
 
+    template <::std::size_t I>
+    constexpr auto
+    get() & -> decltype(get_impl<I>()(*this))
+    {
+        return get_impl<I>()(*this);
+    }
+
+    template <::std::size_t I>
+    constexpr auto
+    get() const& -> decltype(get_impl<I>()(*this))
+    {
+        return get_impl<I>()(*this);
+    }
+
+    template <::std::size_t I>
+    constexpr auto
+    get() && -> decltype(get_impl<I>()(::std::move(*this)))
+    {
+        return get_impl<I>()(::std::move(*this));
+    }
+
+    template <::std::size_t I>
+    constexpr auto
+    get() const&& -> decltype(get_impl<I>()(::std::move(*this)))
+    {
+        return get_impl<I>()(::std::move(*this));
+    }
+
     tuple() = default;
     tuple(const tuple& other) = default;
+    tuple(tuple&& other) = default;
     template <typename U1, typename... U>
     tuple(const tuple<U1, U...>& other) : holder(other.holder.value), next(other.next)
```

Comment:
Potentially we may have overload with `tuple<U1, U...>&&`

Reply:
We don't need it for now, do we? Is it ok for tests?

Reply:
Of course it's ok for tests. The question (as always) do we want to support rvalue category and thus, support move_iterator. Perhaps, this is not a big issue now (or even in long term) because our main use-case is a tuple of references. It doesn't matter if we move them or not.

Reply:
But even with move iterator, will not we have `tuple<U&&...>` and not `tuple<U...>&&`?

Reply:
Thanks for that comment. I've changed my mind:) Seems like we need this constructor. 
You are right that it would be `tuple<U&&...>` and not `tuple<U...>&&` from the `tuple` type perspective but cannot "move from" lvalue. ``tuple`` works in such way that is can move when both tuple is rvalue and it contains rvalue, thus, moving from `tuple` requires to be either ``tuple<U&&>&&`` or `tuple<U&&>`. When `tuple` matches on `const` lvalue reference we lose the information about `tuple` value category.